src/core/navigation/navigation.js

```diff
@@ -114,102 +115,214 @@ export class Navigation {
 		return cell.model();
 	}
 
+	context(type, settings) {
+		const model = this.model;
+		const oldRow = this.currentRow;
+		const oldColumn = this.currentColumn;
+		const keyCode = model.action().shortcut.keyCode;
+
+		return Object.assign({
+			model,
+			type,
+			oldRow,
+			oldColumn,
+			keyCode
+		}, settings);
+	}
+
 	get commands() {
 		const model = this.model;
 		const table = this.table;
 		const shortcut = model.navigation().shortcut;
+		const edit = model.edit;
+
 		const canNavigate = () => {
-			if (model.edit().state === 'view') {
+			if (edit().state === 'view') {
 				return true;
 			}
 
 			const column = table.body.column(this.currentColumn).model();
 			return column && (column.editorOptions.trigger === 'focus' || column.editorOptions.cruise === 'transparent');
 		};
 
+		const go = this.model.navigation().go;
+
 		const commands = {
 			goDown: new Command({
 				shortcut: shortcut.down,
-				canExecute: () => canNavigate() && this.nextRow >= 0,
-				execute: () => this.goTo(this.nextRow, this.currentColumn)
+				canExecute: () => {
+					if (canNavigate()) {
+						const newRow = this.nextRow;
+						return newRow >= 0 && go.canExecute(this.context('down', {newRow}));
+					}
+
+					return false;
+				},
+				execute: () => {
+					const newRow = this.nextRow;
+					const newColumn = this.currentColumn;
+					return go.execute(this.context('down', {newRow, newColumn})) && this.goTo(newRow, newColumn);
```

Comment:
it's minor, but we still have incorrect interface for ```execute``` method in d.ts file which returns void.